IBAnimatable/Animatable.swift

```diff
@@ -207,253 +123,172 @@ public extension Animatable where Self: UIView {
-      animation.duration = CFTimeInterval(self.duration)
-      animation.beginTime = CACurrentMediaTime() + CFTimeInterval(self.delay)
-      animation.autoreverses = true
-      self.layer.add(animation, forKey: "fade")
-    }, completion: completion)
+  public func fade(way: AnimationType.FadeWay, completion: AnimatableCompletion? = nil) {
+    switch way {
+    case .outIn:
+      fadeOutIn(completion)
+    case .inOut:
+      fadeInOut(completion)
+    case .in:
+      alpha = 0
+      animateIn(0, 0, 1, 1, 1, completion)
+    case .out:
+      alpha = 1
+      animateOut(0, 0, 1, 1, 0, completion)
+    }
   }
   
-  public func fadeInOut(_ completion: AnimatableCompletion? = nil) {
-    CALayer.animate({
-      let animation = CABasicAnimation(keyPath: "opacity")
-      animation.fromValue = 0
-      animation.toValue = 1
-      animation.timingFunction = CAMediaTimingFunction(name: kCAMediaTimingFunctionEaseInEaseOut)
-      animation.duration = CFTimeInterval(self.duration)
-      animation.beginTime = CACurrentMediaTime() + CFTimeInterval(self.delay)
-      animation.autoreverses = true
-      animation.isRemovedOnCompletion = false
-      self.layer.add(animation, forKey: "fade")
-    },
-    completion: {
-      self.alpha = 0
-      completion?()
-    })
+  public func squeezeFade(way: AnimationType.Way, direction: AnimationType.Direction, completion: AnimatableCompletion? = nil) {
+    
+    let values = computeValues(way: way, direction: direction, shouldScale: true)
+    switch way {
+    case .in:
+      alpha = 0
+      animateIn(values.x, values.y, values.scaleX, values.scaleY, 1, completion)
+    case .out:
+      animateOut(values.x, values.y, values.scaleX, values.scaleY, 0, completion)
+    }
   }
   
-  public func zoomIn(_ completion: AnimatableCompletion? = nil) {
+  public func zoom(way: AnimationType.Way, completion: AnimatableCompletion? = nil) {
     let scaleX = 2 * force
     let scaleY = 2 * force
-    alpha = 0
-    let toAlpha: CGFloat = 1
-    animateInWithScaleX(scaleX, scaleY: scaleY, alpha: toAlpha, completion: completion)
+    let toAlpha: CGFloat
+    
+    switch way {
+    case .in:
+      alpha = 0
+      toAlpha = 1
+      animateIn(0, 0, scaleX, scaleY, toAlpha, completion)
+    case .out:
+      alpha = 1
+      toAlpha = 0
+      animateOut(0, 0, scaleX, scaleY, toAlpha, completion)
+    }
+    
   }
   
-  public func zoomOut(_ completion: AnimatableCompletion? = nil) {
-    let scaleX = 2 * force
-    let scaleY = 2 * force
-    alpha = 1
-    let toAlpha: CGFloat = 0
-    animateOutWithScaleX(scaleX, scaleY: scaleY, alpha: toAlpha, completion: completion)
-  }
-
-  public func zoomInvertIn(_ completion: AnimatableCompletion? = nil) {
-    let scaleX = 1 * force
-    let scaleY = 1 * force
-    alpha = 0
-    let toAlpha: CGFloat = 1
-    transform = CGAffineTransform(scaleX: 0.1, y: 0.1)
-    animateOutWithScaleX(scaleX, scaleY: scaleY, alpha: toAlpha, completion: completion)
```

Comment:
I didnt test this part. 
It should have been `animateInWithScaleX...` ? @tbaranes